include/cudaq/Optimizer/CodeGen/QIRFunctionNames.h

```diff
@@ -103,4 +103,8 @@ inline mlir::Type getCharPointerType(mlir::MLIRContext *context) {
 
 void initializeTypeConversions(mlir::LLVMTypeConverter &typeConverter);
 
+std::pair<std::size_t, std::vector<std::size_t>>
```

Comment:
I think this needs some documentation here in the header file. I'm not sure what it does just from the name and signature.

Is this a QIR defined function?

Reply:
Will add, thanks! This is a helper to get argument struct size and argument offsets from a func op - we currently use it in several places (quake synthesis tests, python alt_launch_kernel, quake synthesis, BaseRestRemoteClient.h). Let me know if there is a better place for it!

Reply:
done!